### ISSUE TITLE
Plenty of HttpCookie caused request blocking

### ISSUE THREAD

**Issue by zhaowenxishi** [eclipse/jetty.project#7226]
**Jetty version(s)**
    9.4.43.v20210629
**Java version/vendor** `(use: java -version)`
   openjdk 8u302
**OS type/version**

**Description**
  
1、Phenomenon
(1) at begenning:  Many TimeoutException
(2) **** Max requests queued per destination 1024 exceeded for HttpDestination ****

2、We dump the  memory and find:
(1) Almost all of 200 threads are waiting for the lock
![image](https://user-images.githubusercontent.com/18715523/145002651-9714fcad-246d-40c7-a7aa-67a5a14986c4.png)
![image](https://user-images.githubusercontent.com/18715523/145003655-cbe0ec77-e4da-49b8-b039-702d7a2730c7.png)

(2) Except one threads dealing with cookie get
![image](https://user-images.githubusercontent.com/18715523/145003861-160c4867-cb3e-4fd9-8a3c-4226a99f9bb8.png)

(3)we find plenty of httpcookie which is hugely different from the normal situation
![image](https://user-images.githubusercontent.com/18715523/145005537-f9d74b5d-b2e0-4478-ac57-25ec5de3685c.png)

![image](https://user-images.githubusercontent.com/18715523/145005635-0663a008-2da8-4bb3-8654-1485f12c93e6.png)
![image](https://user-images.githubusercontent.com/18715523/145005809-580e4666-ed20-41ac-ab0a-0c0434c54657.png)

3、Point of doubt：
（1）huge sizes of cookie caused the "normalizeRequest of HttpConnection" slowly and then caused the request blocking

**How to reproduce?**
unfortunately，It's an Occasional problem.

the only difference of the normal situation is that:
we may sometime send concurrent multiple requests at the same time

**Comment by zhaowenxishi** [comment]
nothing deal with cookie

**Comment by zhaowenxishi** [comment]
After reanalysis, I found that the problem is still due to this reason, but I can’t confirm why the cookie causes the request to block, so I hope to get some help

**Comment by zhaowenxishi** [comment]
1、here is blocked about 28s before starting sending
![image](https://user-images.githubusercontent.com/18715523/147759651-d7be618f-d94a-4191-8647-04e4627dbceb.png)

2、here is blocked about 32s after receiving. finnally it success but is already timeout for Upper business process。
![image](https://user-images.githubusercontent.com/18715523/147759833-fa42bcdc-2fb3-403e-a98c-098e4d70f1b2.png)

**Comment by zhaowenxishi** [comment]
![image](https://user-images.githubusercontent.com/18715523/147760786-d850ed9e-d141-4780-a241-49aa9517780a.png)

**Comment by lachlan-roberts** [comment]
This is just how the `java.net.InMemoryCookieStore` works, it grabs the lock for each `get` to fill up the list of cookies for a URI, so if you have a large amount of cookies it could be a bottleneck.

It would help if you can reduce the number of cookies you are storing.

**Comment by zhaowenxishi** [comment]
> This is just how the `java.net.InMemoryCookieStore` works, it grabs the lock for each `get` to fill up the list of cookies for a URI, so if you have a large amount of cookies it could be a bottleneck.
> 
> It would help if you can reduce the number of cookies you are storing.

Thanks for your asking. Once I found that problem, I had tried to put a lot of cookies in the java.net.InMemoryCookieStore(even more than the actual situation, 2000+), but I won't work out to reproduce the problem situation.

I am not sure if the my cookies are to simple or most of them have same inner information will affect the result and the InMemoryCookieStore Processing Efficiency.

But I read the source code of InMemoryCookieStore,.When it's dealing with "_it grabs the lock for each get to fill up the list of cookies for a URI_" as you mentioned above, it just do something about "comparing and judging if the information is equal". I don't thing this kind of handling will consume a lot of time, although the size of cookie list is large.
